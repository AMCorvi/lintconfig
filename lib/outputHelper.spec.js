const config = require('../index.js')
const co = require('./outputHelper.js')

test('"config" import exports expected keys', function () {
  expect(config).toEqual(
    expect.objectContaining({
      rules:         expect.any(Object)
      ,env:           expect.any(Object)
      ,parserOptions: expect.any(Object)
      ,plugins:       expect.any(Array)
    })
  )
})

test('Produce array of strings', function () {
  let func = co.__createKeyStrings(Object.keys(config))
  let arrayOfString = [expect.any(String)]
  expect(func).toEqual(expect.arrayContaining(arrayOfString))
  expect(func).toMatchSnapshot()
})

test(".configText is string", () => {
  expect(co.configText).toEqual(expect.any(String))
} )
