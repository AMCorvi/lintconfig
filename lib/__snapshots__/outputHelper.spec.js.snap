// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Produce array of strings 1`] = `
Array [
  "AMConfig.env = Object.assign(AMConfig.env, {
          /* Insert Overriding Eslint env Here */
})
",
  "AMConfig.parserOptions = Object.assign(AMConfig.parserOptions, {
          /* Insert Overriding Eslint parserOptions Here */
})
",
  "AMConfig.plugins = Object.assign(AMConfig.plugins, {
          /* Insert Overriding Eslint plugins Here */
})
",
  "AMConfig.rules = Object.assign(AMConfig.rules, {
          /* Insert Overriding Eslint rules Here */
})
",
]
`;
